core/templates/dev/head/pages/topics_and_skills_dashboard/SkillsListDirective.js

```diff
@@ -129,6 +136,64 @@ oppia.directive('skillsList', [
               }
             });
           };
+
+          $scope.selectSkill = function(skill) {
+            $scope.selectedSkill = skill;
+          };
+
+          $scope.mergeSkill = function(skill) {
+            var skillSummaries = $scope.getMergeableSkillSummaries();
+            var modalInstance = $uibModal.open({
+              templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                '/pages/topics_and_skills_dashboard/' +
+                'merge_skill_modal_directive.html'),
+              backdrop: true,
+              controller: [
+                '$scope', '$uibModalInstance',
+                function($scope, $uibModalInstance) {
+                  $scope.skillSummaries = skillSummaries;
+                  $scope.selectedSkill = {};
+                  $scope.done = function() {
+                    $uibModalInstance.close(
+                      {skill: skill,
+                        supersedingSkill: $scope.selectedSkill
```

Comment:
If you are just using the superseding skill ID, why not pass just the ID instead?

Reply:
Ah, but I think the code here is where the user closes the modal and communicates the selected skill ID to the outside world, right? For that, just the ID should suffice? (Better to pass around minimal stuff, in general.)

Reply:
Done